lib/commands/certificate.js

```diff
@@ -0,0 +1,188 @@
+import _ from 'lodash';
+import { fs, plist } from 'appium-support';
+import { iosCommands } from 'appium-ios-driver';
+import { openUrl } from 'node-simctl';
+import { waitForCondition } from 'asyncbox';
+import B from 'bluebird';
+import log from '../logger';
+import os from 'os';
+import path from 'path';
+import UUID from 'uuid-js';
+import x509 from 'x509';
+
+
+let extensions = {}, commands = {};
+
+const SELF_MODULE_NAME = 'appium-xcuitest-driver';
+const BASE_MODULE_NAME = 'appium-base-driver';
+const CONFIG_EXTENSION = 'mobileconfig';
+
+
+async function toMobileConfig (certBuffer) {
+  let subject = {};
+  try {
+    subject = x509.getSubject(certBuffer.toString('ascii'));
+  } catch (err) {
+    throw new Error(`Cannot parse the certificate. Is it valid and base64-encoded? ` +
+                    `Original error: ${err.message}`);
+  }
+  const displayName = _.replace(subject.commonName, /\*/g, '_');
+  if (_.isEmpty(displayName)) {
+    throw new Error(`Cannot parse commonName field from the certificate subject. ` +
+                    `Original value: ${JSON.stringify(subject)}`);
+  }
+  const getUUID = () => UUID.create().hex.toUpperCase();
+  const contentUuid = getUUID();
+  return {
+    PayloadContent: [{
+      PayloadCertificateFileName: `${displayName}.cer`,
+      PayloadContent: certBuffer,
+      PayloadDescription: 'Adds a CA root certificate',
+      PayloadDisplayName: displayName,
+      PayloadIdentifier: `com.apple.security.root.${contentUuid}`,
+      PayloadType: 'com.apple.security.root',
+      PayloadUUID: contentUuid,
+      PayloadVersion: 1
+    }],
+    PayloadDisplayName: displayName,
+    PayloadIdentifier: `${os.hostname().split('.')[0]}.${getUUID()}`,
+    PayloadRemovalDisallowed: false,
+    PayloadType: 'Configuration',
+    PayloadUUID: getUUID(),
+    PayloadVersion: 1
+  };
+}
+
+async function clickElement (driver, locator, timeout = 5000) {
+  let element = null;
+  try {
+    await waitForCondition(async () => {
```

Comment:
`retryInterval` was made for these sorts of situations, where an exception is thrown:
```js
try {
  element = await retryInterval(timeout / 500, 500, async () => {
    return await driver.findNativeElementOrElements(locator.type, locator.value, false);
  });
} catch (err) {
  throw new Error(`Cannot find ${JSON.stringify(locator)} within ${timeout}ms timeout`);
}
```